File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/go_solutions/216-combination-sum-iii.go
func combinationSum3(k int, n int) [][]int {
    var res [][]int
    var backtrack func(start, sum int, path []int)
    backtrack = func(start, sum int, path []int) {
        if len(path) == k && sum == n {
            tmp := make([]int, k)
            copy(tmp, path)
            res = append(res, tmp)
            return
        }
        if len(path) > k || sum > n {
            return
        }
        for i := start; i <= 9; i++ {
            path = append(path, i)
            backtrack(i+1, sum+i, path)
            path = path[:len(path)-1]
        }
    }
    backtrack(1, 0, []int{})
    return res
}

File: leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/216-combination-sum-iii.cpp
class Solution {
public:
    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> res;
        vector<int> path;
        backtrack(res, path, 1, k, n);
        return res;
    }

    void backtrack(vector<vector<int>>& res, vector<int>& path, int start, int k, int n) {
        if (path.size() == k && n == 0) {
            res.push_back(path);
            return;
        }
        for (int i = start; i <= 9; ++i) {
            if (n - i < 0) break;
            path.push_back(i);
            backtrack(res, path, i + 1, k, n - i);
            path.pop_back();
        }
    }
};